### PR TITLE
auto renewals: add automatic_renewal task

### PR DESCRIPTION
**Pull request by PascalRepond**
* Adds circ_policy field to set automatic renewals.
* Adds basic task to automatically renew loans on their due date.

* TODO: 
    * write more tests
    * check that a non-renewable loan is not renewed and does not exit the task
    * add notification

### CODE REVIEW

File: rero_ils/modules/loans/api.py [hunk 1]
```diff
@@ -1251,6 +1251,40 @@ def get_overdue_loans(patron_pid=None, tstamp=None):
         yield Loan.get_record_by_pid(pid)
 
 
+def get_loans_pids_by_due_date(tstamp=None):
+    """Return the pids of all loans that are due on a given date.
+
+    :param tstamp: a timestamp to define the due date to filter by. Default is
+    now.
+    :return: a generator of loan pids.
```
Comment (jma) [inline rero_ils/modules/loans/api.py:1259]: This is wrong.

File: rero_ils/modules/loans/tasks.py [hunk 2]
```diff
@@ -47,6 +49,23 @@ def loan_anonymizer(dbcommit=True, reindex=True):
     return counter
 
 
+@shared_task(ignore_result=True)
+def automatic_renewal(tstamp=None):
+    """Extend all loans with an automatic renewal policy."""
```
Comment (jma) [inline rero_ils/modules/loans/tasks.py:54]: Docstring is not complete.

File: rero_ils/modules/loans/tasks.py [hunk 3]
```diff
@@ -47,6 +49,23 @@ def loan_anonymizer(dbcommit=True, reindex=True):
...
+    loans_to_extend = []
+    extended_loans_count = 0
+    for loan in get_loans_by_due_date(tstamp):
+        policy = get_circ_policy(loan)
+        if policy['automatic_renewal']:
+            item = Item.get_record(loan.item_pid)
+            loans_to_extend.append({"loan": loan, "item": item})
+    for loan_item in loans_to_extend:
```
Comment (jma) [inline rero_ils/modules/loans/tasks.py:62]: Why do you need two loops?

**Review by jma — CHANGES_REQUESTED** [review]
tests are missing

**Review by rerowep — APPROVED** [review]

**Pull request closed without merging**